### ISSUE TITLE
Investigate whether we have the API endpoints we'd need in profile to allow incremental search of students by name

### ISSUE THREAD

**Issue by floehopper** [RaspberryPiFoundation/editor-api#345]
This relates to the planned incremental search box on the "Add class members" modal - see [the designs](https://www.figma.com/design/SrINjD1MCT6VWYf1Yc9fV1/Code-Editor---Designs?node-id=7007-52100&t=96ngnU43qCkwfn4K-0).

There was some concern that we might need to make an HTTP request to `editor-api` / `profile` every time the text in the search box changed and or perhaps even an HTTP request for each student.

**Comment by floehopper** [comment]
I think that even at the time this was raised, we thought that we could load all the students for a school in a single request to `editor-api` and then do the filtering on the client side.

I have confirmed that the `Api::ClassMembersController#index` action calls `ClassMember.with_students` which calls `ClassMember.students` which calls `User.from_userinfo` with an array of IDs. The latter returns the student's name and the results are combined into `@class_members_with_students` in the controller so the student names can be rendered in the JSON in `app/views/api/class_members/index.json.jbuilder`.  See code snippets below.

Anyway, in summary, I think we should be fine and we shouldn't need any new endpoints in the Profile app.

https://github.com/RaspberryPiFoundation/editor-api/blob/6e4f9bbf746ada253d06d490f192a4ee46663ef2/app/controllers/api/class_members_controller.rb#L10-L13

https://github.com/RaspberryPiFoundation/editor-api/blob/6e4f9bbf746ada253d06d490f192a4ee46663ef2/app/models/class_member.rb#L18-L21

https://github.com/RaspberryPiFoundation/editor-api/blob/6e4f9bbf746ada253d06d490f192a4ee46663ef2/app/models/class_member.rb#L14-L16

https://github.com/RaspberryPiFoundation/editor-api/blob/6e4f9bbf746ada253d06d490f192a4ee46663ef2/app/models/user.rb#L61-L70

https://github.com/RaspberryPiFoundation/editor-api/blob/6e4f9bbf746ada253d06d490f192a4ee46663ef2/app/views/api/class_members/index.json.jbuilder#L14

**Comment by sra405** [comment]
I think we've missed a step. When students are created we fire off an event to profile and get the IDs returned, currently we do nothing with these IDs. When we come to add a student to a class we don't know what students are in that school (currently scoped by the school code in profile). I do agree once a student is added to a class we can get their details but its the interim step of getting a list of students in the school to select which ones we want to add to the class I think we're missing.

I've looked at the docs and a new endpoint seems to have been added which returns the student IDs of a given school ID [here](https://my.raspberrypi.org/api/v1/documentation/#/Students/post_schools__schoolId__students_list) 🙌 

So I think with:
- the above to get the details of students when attached to a class
- the profile endpoint to return an array of student IDs
I think we're covered!
